### PR TITLE
Longest Side Audit for archery.json, 500.json, 22.json, 38.json and atgm.json

### PR DESCRIPTION
**Pull request by venum4k**
#### Summary

SUMMARY: Content "Added and checked lengths for several guns, bows and arrows"

#### Purpose of change

Addresses some of the issues with item longest_side being inaccurate, assisting with #41691 and the bows for #43420.

#### Describe the solution

I added some longest_side lines to jsons and changed some numbers.

#### Describe alternatives you've considered

I'm not sure there are any real alternatives to inputting numbers

#### Testing

I checked the in-game inventory screens for updated values.

#### Additional context

A lot of the bow lengths were taken from the existing descriptions in the json files. Some of the gun lengths were from manufacturer websites, some educated guesses were made (hey, the numbers are probably more accurate than they were before).

### CODE REVIEW

**Comment by venum4k** [comment]
Attempt #2 cause I haven't done git in a while and apparently I'm dumb

**Comment by LaVeyanFiend** [comment]
longest_side is based on Feret distance, not OAL. All the gun lengths will be too short. Please use the tutorial on #37374.

**Comment by venum4k** [comment]
I had a hunch that it wasn't just the length, I've tried out the tutorial on the Big Horn 89.
image
S&W 500
image
Ruger LCR 22
image
Ruger LCR 38 - 0.2mm off from the previous length that I thought was wrong. 
image
BGM-71F TOW - Couldn't find a good image for the F variant but I doubt the Feret distance would be any longer. I haven't included the tripod.
image
Sig Mosquito
image
That should be all the guns I've modified. Bow length should be the same as their Feret distance from what I understand.

File: data/json/items/gun/500.json [hunk 1]
```diff
@@ -8,6 +8,7 @@
     "description": "Built in the image of the venerable Winchester Model 1886, Big Horn Armory's Model 89 rifle packs all the power of the .500 S&W Magnum cartridge in to a package less likely to break one's wrist.",
     "weight": "3447 g",
     "volume": "3 L",
+    "longest_side": "940 mm",
```
Comment (LaVeyanFiend) [inline data/json/items/gun/500.json]:
```suggestion
    "volume": "1995 ml",
    "longest_side": "953 mm",
```
Listing of the gun with 18" configuration as you chose for the longest side with views on all sides: https://www.gunsinternational.com/guns-for-sale-online/rifles/big-horn-armory/big-horn-armory---89----500-s-w.cfm?gun_id=101297785#lg-1
Extraction of gun from image for scaling: https://imgur.com/a/HAU8mtY
SA+Feret: https://imgur.com/a/3IxR2ni
Measuring width of grip cap to match to the bottom view of the rifle to get width: https://imgur.com/a/wERhTUn
Width after scaling to grip cap's width from bottom view of gun: https://imgur.com/a/plqdfrx

**Review by LaVeyanFiend — CHANGES_REQUESTED** [review]

File: data/json/items/gun/500.json [hunk 2]
```diff
@@ -6,8 +6,9 @@
     "type": "GUN",
     "name": { "str_sp": "Big Horn Model 89" },
     "description": "Built in the image of the venerable Winchester Model 1886, Big Horn Armory's Model 89 rifle packs all the power of the .500 S&W Magnum cartridge in to a package less likely to break one's wrist.",
-    "weight": "3447 g",
+    "weight": "1995 g",
     "volume": "3 L",
```
Comment (LaVeyanFiend) [inline data/json/items/gun/500.json:10]:
```suggestion
    "weight": "3742 g",
    "volume": "1995 ml",
```
Swapped the field to change, whoops :3
(The weight on Big Horn's website is for the 16" model, I got this from the listing I used for volume+length calculations)

**Comment by ZhilkinSerg** [comment]
Apparently it fails some ranged balance tests now.

**Comment by venum4k** [comment]
As I've only recently started adding things to cdda, I'm not really sure what the ranged balance tests are. What do they do and how would I address failing them?

**Comment by kevingranade** [comment]
The failure says you can no longer load arrows into bows, you may need to update the pocket data for the bows to make that work again.

**Comment by kevingranade** [comment]
Had a chance to look at this, it turns out the problem is in the test itself.
The arrow is now too long to fit in the character/s backpack, so the arrow displaces the already wielded bow, and since the wielded arrow is not itself reloadable, much less with other errors, the test rightfully throws it's hands up and goes to have a beer.

**Pull request merged**